citam/engine/calculators/close_contacts_calculator.py

```diff
@@ -0,0 +1,377 @@
+        with open(filename, "w") as json_file:
+            json.dump(statistics_dict, json_file)
+
+        # Per-floor outputs
+        for floor_number in range(self.facility.number_of_floors):
+
+            # Distribution of contact locations
+
+            floor_directory = os.path.join(
+                work_directory, "floor_" + str(floor_number)
+            )
+
+            coord_contact_dist_filename = os.path.join(
+                floor_directory, "contact_dist_per_coord.csv"
+            )
+
+            contacts_per_coord = (
+                self.contact_events.get_contacts_per_coordinates(floor_number)
+            )
+
+            with open(coord_contact_dist_filename, "a") as outfile:
+                outfile.write("x,y,n_contacts\n")
+                for pos in contacts_per_coord:
+                    x, y = pos
+                    dur = contacts_per_coord[pos]
+                    outfile.write(
+                        str(x) + "," + str(y) + "," + str(dur) + "\n"
+                    )
+
+            self.create_svg_heatmap(contacts_per_coord, floor_directory)
+
+    def create_svg_heatmap(
+        self,
+        contacts_per_coord: Dict[Tuple[int, int], int],
+        floor_directory: str,
+    ) -> None:
+        """
+        Create and save a heatmap from coordinate contact data.
+
+        :param contacts_per_coord: dictionary where each key is an (x,y )
+                tuple and values are the number of contacts in that location.
+        :type contacts_per_coord: Dict[Tuple[int, int], int]
+        :param floor_directory: directory to save the heatmap file.
+        :type floor_directory: str
+        :raises FileNotFoundError: if the floor svg map file is not found.
+        """
+
+        heatmap_file = os.path.join(floor_directory, "heatmap.svg")
+        map_file = os.path.join(floor_directory, "map.svg")
+
+        if not os.path.isfile(map_file):
+            raise FileNotFoundError(map_file)
+
+        max_contacts = 0
+        if len(contacts_per_coord.values()) > 0:
+            max_contacts = max(contacts_per_coord.values())
+        tree = ET.parse(map_file)
+
+        root = tree.getroot()[0]
+
+        filter_elem = ET.Element("filter")
+        filter_elem.set("id", "blurMe")
+        filter_elem.set("width", str(self.contact_distance * 2))
+        filter_elem.set("height", str(self.contact_distance * 2))
+        filter_elem.set("x", "-100%")
+        filter_elem.set("y", "-100%")
+
+        blur_elem = ET.Element("feGaussianBlur")
+        blur_elem.set("in", "SourceGraphic")
+        blur_elem.set("stdDeviation", str(self.contact_distance * 0.4))
+        blur_elem.set("edgeMode", "duplicate")
+
+        filter_elem.append(blur_elem)
+        root.append(filter_elem)
+        color_scale = cm.get_cmap("RdYlGn")
+
+        for key, v in contacts_per_coord.items():
+            x, y = key
+            color = color_scale(1.0 - v * 1.0 / max_contacts)
```

Comment:
Potential divide by zero exception here if  `max_contacts` is not updated in `if` block at line# 344

Reply:
This is actually correct. If max_contacts is zero, this means there was no contact which in turn means contacts_per_coord will be empty and therefore the loop will not be executed.